datafusion/physical-plan/src/joins/sort_merge_join.rs

```diff
@@ -877,11 +976,14 @@ impl SMJStream {
                     }
                     Poll::Ready(Some(batch)) => {
                         if batch.num_rows() > 0 {
+                            //dbg!("streamed poll ready");
                             self.freeze_streamed()?;
                             self.join_metrics.input_batches.add(1);
                             self.join_metrics.input_rows.add(batch.num_rows());
                             self.streamed_batch =
                                 StreamedBatch::new(batch, &self.on_streamed);
+                            self.streamed_batch_counter
```

Comment:
we need to track batch ids, to prevent edge case like
Batch 0 value 1 streamed index 0
Batch 1 value 2 streamed index 0

So the streamed index is the same, and without batch id its not possible to figure out the actual values are different

Reply:
Can you please put that context as comments of the docs of `stream_batch_counter`?